hpat/hiframes/boxing.py

```diff
@@ -154,12 +154,21 @@ def _infer_series_list_dtype(S):
 
 
 def _infer_index_type(index):
-    # TODO: support proper inference
+    '''
+    Convertion input index type into Numba known type
+    need to return instance of the type class
+    '''
+
+    if isinstance(index, (types.NoneType, pd.RangeIndex)) or index is None:
```

Comment:
But the following (existing) tests will still fail if we don't create index from pd.RangeIndex during unboxing:

```
    def test_series_dropna_float1(self):
        def test_impl(A):
            return A.dropna()
        hpat_func = hpat.jit(test_impl)

        S1 = pd.Series([1.0, 2.0, np.nan, 1.0, np.inf])
        S2 = S1.copy()
        pd.testing.assert_series_equal(hpat_func(S1), test_impl(S2))
```
The error will be:
Series.index values are different (50.0 %)
[left]:  RangeIndex(start=0, stop=4, step=1)
[right]: Int64Index([0, 1, 3, 4], dtype='int64')


```
    def test_series_default_index(self):
        def test_impl():
            A = pd.Series([3, 2, 1, 5, 4])
            return A.index

        hpat_func = hpat.jit(test_impl)
        np.testing.assert_array_equal(hpat_func(), test_impl())
```
The error will be:
AssertionError:
Arrays are not equal

Mismatch: 500%
 x: array(None, dtype=object)
 y: array([0, 1, 2, 3, 4], dtype=int64)

Reply:
@shssf Yes, because existing version of test_series_dropna_float1 omits checking Series indexes at all.
Please see that the assert in the code I provided is different:
`pd.testing.assert_series_equal(hpat_func(S1), test_impl(S2))`

Reply:
Also one more test that don't seem to work with current design:

```
    def test_pass_series_int_index(self):
        '''Verifies boxing and unboxing of a Series with integer index'''
        def test_impl(A):
            return A
        hpat_func = hpat.jit(test_impl)

        S = pd.Series([3, 5, 6], [1, 2, 3], name='A')
        pd.testing.assert_series_equal(hpat_func(S), test_impl(S))
```
Also please use the following git patch to add more tests (including the one above), remove duplicate test and add docstrings: [patch]( https://github.com/kozlov-alexey/hpat/blob/e05ed780cf32311f27e506004fac8762654eb80e/index_tests_patch.txt)